### PR TITLE
Constant extension properties no longer use convention mapping

### PR DESCRIPTION
**Pull request by Crazyjavahacking**

### CODE REVIEW

File: src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy [hunk 1]
```diff
@@ -108,9 +108,9 @@ class ExcelsiorJetPlugin implements Plugin<Project> {
             }
         }
         extension.conventionMapping.jetHome = { System.getProperty("jet.home") }
-        extension.conventionMapping.jetResourcesDir = {
-            project.file("src/main/jetresources")
-        }
```
Comment (Crazyjavahacking) [inline src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy:113]: Here it is not necessary to use convention mapping. The assignment is effectively a constant as `Project.file()` will always return the same path once Gradle's initialization phase is done.

File: src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy [hunk 2]
```diff
@@ -123,13 +123,8 @@ class ExcelsiorJetPlugin implements Plugin<Project> {
             }
         }
 
-        extension.conventionMapping.appType = {
-            if (isWar)  {
-                return ApplicationType.TOMCAT.toString()
-            } else {
-                return ApplicationType.PLAIN.toString();
-            }
-        }
```
Comment (Crazyjavahacking) [inline src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy:132]: The same applies here. This is constant assignment as isWar is evaluated when plugin is applied.

File: src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy (hunk 1 above)
Comment (pjBooms) [reply src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy:113]: But a plugin user can change the jetResoursesDir property. AFAIK, the whole idea of convention mapping is to use convention values when the user does not change the property. 
If we just assign the property here won't it overwrite user's value?

Comment (Crazyjavahacking) [reply src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy:113]: The main idea of convention mapping was to **solve the evaluation ordering problem**.

So if the evaluation happens in configuration phase, but some other task or plugin modifies the value in execution phase, the plugin will use old and most likely no longer valid value.

So for example the property `jetHome` cannot be simplified in similar way as potentially some other plugin might modify the system property at runtime (execution phase).

There should not be any problem with further overwriting the value of `jetResourcesDir` property as it is a normal Groovy object property. And it should also be safe to refactor as the `Project.file()` call will always evaluate to the same value.

Comment (Crazyjavahacking) [reply src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy:113]: We should really add some integration tests to guarantee everything works as before.

**Pull request merged**